documentation/src/main/asciidoc/elasticsearch-integration.asciidoc

```diff
@@ -757,6 +751,63 @@ This interface and `FieldBridge` in general are likely going to evolve in the ne
 to remove its adherence to Lucene specific classes like `Document`.
 --
 
+====== Sub-fields
+
+When your `MetadataProvidingFieldBridge` registers a field whose name is the name of an existing field,
+with a dot and another string appended, like `name + ".mySubField"`,
+Hibernate Search will translate it as an object with a property `mySubField`
+in the JSON document sent to Elasticsearch.
+
+As a result, sub-fields can only have a parent field with the `OBJECT` type:
+obviously, Elasticsearch would reject a String or an Integer with `mySubField` property.
+So every time a field named `foo.bar` is registered,
+its parent field `foo` must be registered with the `OBJECT` type, as in the following example.
+Not doing so will result in errors when Hibernate Search generates the Elasticsearch schema.
+
+[source,java]
+--
+@Override
+public void configureFieldMetadata(String name, FieldMetadataBuilder builder) {
+    builder
+        .field( name, FieldType.OBJECT )
+        .field( name + ".firstName", FieldType.STRING )
+            .sortable( true )
+        .field( name + ".middleName", FieldType.STRING )
+            .sortable( true );
+}
+--
+
+[[elasticsearch-custom-field-bridge-dynamic-mapping]]
+====== Dynamic mapping
+
+By default Hibernate Search (or, more accurately, the Elasticsearch schema generated by Hibernate Search)
+will not allow custom bridges to create fields that weren't
+previously registered through the `configureFieldMetadata` method.
+
+If you don't know in advance the name of the fields created by your custom bridge,
+you can enable "dynamic mapping" globally or per-index through <<elasticsearch-integration-configuration,a configuration option>>,
+or you can do it on a per-field basis using the `FieldMetadataCreationContext.mappedOn` method
+and the `org.hibernate.search.elasticsearch.bridge.spi.Elasticsearch` extension:
+
+[source,java]
+--
+@Override
+public void configureFieldMetadata(String name, FieldMetadataBuilder builder) {
+    builder
+        .field( name, FieldType.OBJECT )
+            .mappedOn( Elasticsearch.class )
+                .dynamic( DynamicType.TRUE ); // Will allow to index fields named <name>.foo, <name>.bar, etc.
+}
+--
+
+[NOTE]
+---
+Dynamic mapping involves Elasticsearch guessing the type and attributes of fields automatically,
+and has its own limitations.
+See https://www.elastic.co/guide/en/elasticsearch/reference/current/dynamic-mapping.html[the reference documentation]
+for more information. 
+---
```

Comment:
something is wrong here, it's not rendering the note but renders the `---` literally..